lute/luau/src/luau.cpp

```diff
@@ -2657,12 +2657,10 @@ int compile_luau(lua_State* L)
 int load_luau(lua_State* L)
 {
     const std::string* bytecode_string = static_cast<std::string*>(luaL_checkudata(L, 1, COMPILE_RESULT_TYPE));
-    const char* path = luaL_optlstring(L, 2, "=luau.load", nullptr);
-    std::string chunk_name = path;
-    if (chunk_name != "=luau.load")
-        chunk_name.insert(0, "@");
+    const char* path = luaL_checkstring(L, 2);
+    int envIndex = lua_isnoneornil(L, 3) ? 0 : 3;
 
-    luau_load(L, chunk_name.c_str(), bytecode_string->c_str(), bytecode_string->length(), lua_gettop(L) > 2 ? 3 : 0);
+    luau_load(L, path, bytecode_string->c_str(), bytecode_string->length(), envIndex);
```

Comment:
```suggestion
    if (luau_load(L, path, bytecode_string->c_str(), bytecode_string->length(), envIndex) != 0)
        luaL_error(L, "could not load bytecode: %s", lua_tostring(L, -1));
```
Sorry, some of this stuff isn't stuff that you missed, it's just been broken and I'm realizing it now.

Reply:
Alternatively, just this:
```suggestion
    if (luau_load(L, path, bytecode_string->c_str(), bytecode_string->length(), envIndex) != 0)
        lua_error(L); // luau_load pushed the error message onto the stack
```